addons/binding/org.openhab.binding.omnikinverter/src/main/java/org/openhab/binding/omnikinverter/OmnikInverterMessage.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.omnikinverter;
+
+import java.nio.ByteBuffer;
+
+/**
+ *
+ * @author Hans van den Bogert - Initial contribution
+ *
+ */
+public class OmnikInverterMessage {
+
+    byte[] bytes;
+
+    public OmnikInverterMessage(byte[] b) {
+        this.bytes = b;
+    }
+
+    public float getPower() {
```

Comment:
More precision never hurts,but I figured since the numbers are 16 bit, a float would be sufficient. But thanks for forcing me to look into this again, since the `getTotalEnergy` is 32 bits and float would be insufficient. Doubles it is!